sound/soc/sof/topology.c

```diff
@@ -1993,11 +2006,15 @@ static int sof_link_ssp_load(struct snd_soc_component *scomp, int index,
 		return ret;
 	}
 
-	/* set config for all DAI's with name matching the link name */
-	ret = sof_set_dai_config(sdev, size, link, config);
-	if (ret < 0)
-		dev_err(sdev->dev, "error: failed to save DAI config for SSP%d\n",
-			config->dai_index);
+	/* set config to dai component of playback */
+	if (link->dpcm_playback)
+		ret = sof_set_dai_config(sdev, size, link, config,
+					 SNDRV_PCM_STREAM_PLAYBACK);
+
```

Comment:
@RanderWang did you intend to proceed further if ret is < 0 here?

Reply:
@ranj063 no, I will refine it

Reply:
@RanderWang the other thing Pierre suggested was to use a loop so you dont have the same code twice for playback and capture. You could try something like this:

for (dir = SNDRV_PCM_STREAM_PLAYBACK; dir <= SNDRV_PCM_STREAM_CAPTURE; dir++) {
		flag = (dir == SNDRV_PCM_STREAM_PLAYBACK) ? link->dpcm_playback : link->dpcm_capture;
		if (flag)
			ret = sof_set_dai_config(sdev, size, link, config, dir);
	}

Reply:
do we need to use loop for two items? I don't like it

Reply:
I am even wondering why we can't deal with the direction inside of sof_set_dai_config since all the other arguments are identical. It'd simplify the code further.